pkg/cmd/inspect.go

```diff
@@ -155,6 +155,93 @@ func (o *InspectOptions) Validate() error {
 	return nil
 }
 
+func objectReferenceToResourceInfo(clientGetter genericclioptions.RESTClientGetter, ref *configv1.ObjectReference) (*resource.Info, error) {
+	b := resource.NewBuilder(clientGetter).
+		Unstructured().
+		ResourceTypeOrNameArgs(false, fmt.Sprintf("%s/%s", ref.Resource, ref.Name)).
+		NamespaceParam(ref.Namespace).
+		Flatten().
+		Latest()
+
+	infos, err := b.Do().Infos()
+	if err != nil {
+		return nil, err
+	}
+
+	return infos[0], nil
+}
+
+func infoToContextKey(info *resource.Info) string {
+	return fmt.Sprintf("%s/%s.%s/%s/%s", info.Namespace, info.ResourceMapping().GroupVersionKind.Group, info.ResourceVersion, info.ResourceMapping().Resource.Resource, info.Name)
+}
+
+// collectPolymorphicResourceInfo receives an object to gather debugging data for, and a context to keep track of
+// already-seen objects when following related-object reference chains.
+func (o *InspectOptions) collectPolymorphicResourceInfo(info *resource.Info, context map[string]bool) ([]string, error) {
```

Comment:
structure like polymorphic subresources where it's a bare method not attached to the options struct